src/main/java/com/microsoft/aad/msal4j/Authority.java

```diff
@@ -36,6 +36,17 @@
     String authorizationEndpoint;
     String tokenEndpoint;
 
+    final static String DEVICE_CODE_ENDPOINT = "oauth2/devicecode";
+
+    private final static String ADFS_AUTHORITY_FORMAT = "https://%s/%s/";
+    private final static String DEVICE_CODE_ENDPOINT_FORMAT = ADFS_AUTHORITY_FORMAT + DEVICE_CODE_ENDPOINT;
```

Comment:
why general authority contains fields from child class ?

Reply:
It was leftover from when I initially started working on it and thought all authorities would now need to the device code endpoint (forgot about B2C authority). In the latest commit I cleaned it up to follow better code practices.